.github/workflows/CICD.yml

```
@@ -0,0 +1,298 @@
+name: Integrate
+
+# Run this workflow every time a new commit pushed to your repository
+on:
+  push:
+    paths-ignore:
+      - '**/*.md'
+      - 'public/dist/*.js'
+      - 'public/dist/**/*.js'
+  pull_request:
+    paths-ignore:
+      - '**/*.md'
+      - 'public/dist/*.js'
+      - 'public/dist/**/*.js'
+  # Allow manually triggering the workflow.
+  workflow_dispatch:
+
+jobs:
+  kill_previous:
+    name: 0️⃣ Kill previous runs
+    runs-on: ubuntu-latest
+    # We want to run on external PRs, but not on our own internal PRs as they'll be run by the push to the branch.
+    if: (github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository)
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.7.0
+        with:
+          access_token: ${{ github.token }}
+
+  php_syntax_errors:
+    name: 1️⃣ PHP - Syntax errors
+    runs-on: ubuntu-latest
+    needs:
+      - kill_previous
+    steps:
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: latest
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install dependencies
+        uses: ramsey/composer-install@v2
+
+      - name: Check source code for syntax errors
+        run: vendor/bin/parallel-lint --exclude .git --exclude vendor .
+
+  code_style_errors:
+    name: 1️⃣ PHP - Code Style errors
+    runs-on: ubuntu-latest
+    needs:
+      - kill_previous
```

```suggestion
    needs:
      - php_syntax_errors
```

Shouldn't code style depend on syntax? I mean what sense does it make to check code style like correct indention if the syntax, e.g. closing and opening braces, are wrong?